tests/integration/ambient/baseline_test.go

```diff
@@ -3569,11 +3573,27 @@ func restartZtunnel(t framework.TestContext) {
 	}, retry.Timeout(60*time.Second), retry.Delay(2*time.Second)); err != nil {
 		t.Fatalf("failed to wait for ztunnel rollout status for: %v", err)
 	}
-	if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(t.AllClusters()[0], i.Settings().SystemNamespace, "app=ztunnel")); err != nil {
+	if _, err := kubetest.CheckPodsAreReady(kubetest.NewPodFetch(t.AllClusters()[0], ztunnelNS, "app=ztunnel")); err != nil {
 		t.Fatal(err)
 	}
 }
 
+// Locate DaemonSet namespace by name
+func locateDaemonsetNS(t framework.TestContext, dsName string) (string, error) {
+	daemonsets, err := t.Clusters().Default().Kube().AppsV1().DaemonSets("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		t.Fatalf("unable to fetch daemonsets list: %v", err)
```

Comment:
It would be better to return an error here instead of using `t.Fatalf`, and let the calling context decide how to handle it.